src/Features/Core/Portable/MoveStaticMembers/MoveStaticMembersWithDialogCodeAction.cs

```diff
@@ -147,5 +154,106 @@ private static TypeKind GetNewTypeKind(ImmutableArray<ITypeParameterSymbol> type
+                // track this node in case the doc has changed
+                docEditor.ReplaceNode(refNode, refNode.TrackNodes(refNode));
+
+                // add imports for each reference as there may be multiple locations that we need to add to
+                // if the import is already there, we shouldn't add it again
+                var addImports = doc.GetRequiredLanguageService<IAddImportsService>();
+                docEditor.ReplaceNode(docEditor.OriginalRoot, (node, generator) => addImports.AddImport(
+                    docEditor.SemanticModel.Compilation,
+                    node,
+                    node.GetCurrentNode(refNode)!,
+                    generator.NamespaceImportDeclaration(
+                        newType.ContainingNamespace.ToDisplayString(SymbolDisplayFormats.NameFormat))
+                        .WithAdditionalAnnotations(Simplification.Simplifier.Annotation, Formatting.Formatter.Annotation),
+                    generator,
+                    true,
+                    doc.CanAddImportsInHiddenRegions(),
+                    cancellationToken));
+
+                // now change the actual references to use the new type name
+                // extension methods do not need to be changed as we do not reference the class name
+                if (isExtension)
+                {
+                    continue;
+                }
+
+                // We add a simplifier annotation in both cases (second case is done within the method)
+                // Because sometimes the qualification is wholly or partly unneccessary
+                // static member access should never be pointer or conditional member access
+                if (syntaxFacts.IsNameOfSimpleMemberAccessExpression(refNode))
+                {
+                    var expression = syntaxFacts.GetExpressionOfMemberAccessExpression(refNode.Parent);
+                    if (expression != null)
+                    {
+                        docEditor.ReplaceNode(expression, (node, generator) => generator.TypeExpression(newType)
+                            .WithTriviaFrom(node)
+                            .WithAdditionalAnnotations(Simplification.Simplifier.Annotation));
+                    }
+                }
+                else if (syntaxFacts.IsIdentifierName(refNode))
+                {
+                    // We now are in an identifier name that isn't a member access expression
+                    // This could either be because of a static using, module usage in VB, or because we are in the original source type
+                    // either way, we want to change it to a member access expression for the type that is imported
+                    docEditor.ReplaceNode(refNode, (node, generator) => generator.MemberAccessExpression(generator.TypeExpression(newType), node).WithTriviaFrom(node));
+                }
```

Comment:
@CyrusNajmabadi Here's what I'm thinking so far in terms of references that FAR can pull up, please let me know if you think I've missed a case or mishandled it here.

1. Extension Method. We don't need to refactor this, as the usage should stay the same. It is invoked on the instance of the first parameter, so as long as we properly import our new class/namespace the usage should be fine.
2. Member access expression. I only go with a simple member access expression here because the other types of member access expression (pointer and conditional) don't apply to static types. In this case we simply need to replace the expression (the type qualifier) and change it to our new type, so something like `Class1.Foo()` -> `Class1Helpers.Foo()`
3. Non-qualified member access, either because of a static using, imported module, or in the class that we just pulled the members out of. In this case, we want replace the current identifier with a member access expression instead, adding our qualifying type to the original name. Example: `Foo()` -> `FooNs.Class1Helpers.Foo()` (which then can be shortened to `Class1Helpers.Foo()` with the simplifier annotation and the import statement).